src/units/abilities.cpp

```diff
@@ -1082,6 +1074,214 @@ namespace { // Helpers for attack_type::special_active()
+	if(filter_student || filter_opponent || filter_attacker || filter_defender){
+		return true;
+	}
+	return false;
+}
+
+
+bool leadership_affects_self(const std::string& ability, const unit &un, const map_location& loc, bool attacker)
+{
+	unit_ability_list abil = un.get_abilities(ability, loc);
+	for(unit_ability_list::iterator i = abil.begin(); i != abil.end();) {
+		const std::string& apply_to = (*i->first)["apply_to"];
+		if(apply_to.empty() || apply_to == "both" || apply_to == "self") {
+			return true;
+		}
+		if(attacker && apply_to == "attacker") {
+			return true;
+		}
+		if(!attacker && apply_to == "defender") {
+			return true;
+		}
+		++i;
+	}
+	return false;
+}
+
+bool leadership_affects_opponent(const std::string& ability, const unit &un, const map_location& loc, bool attacker)
+{
+	unit_ability_list abil = un.get_abilities(ability, loc);
+	for(unit_ability_list::iterator i = abil.begin(); i != abil.end();) {
+		const std::string& apply_to = (*i->first)["apply_to"];
+		if(apply_to == "both" || apply_to == "opponent") {
+			return true;
+		}
+		if(attacker && apply_to == "defender") {
+			return true;
+		}
+		if(!attacker && apply_to == "attacker") {
+			return true;
+		}
+		++i;
+	}
+	return false;
+}
+
+unit_ability_list list_leadership(const std::string& ability, unit_const_ptr un, unit_const_ptr up, const map_location& loc, const map_location& opp_loc, bool attacker, const_attack_ptr weapon, const_attack_ptr opp_weapon, bool affect_self, bool affect_other)
+{
+	unit_ability_list abiln (loc);
+	unit_ability_list abil = (*un).get_abilities(ability, loc);
+	for(unit_ability_list::iterator i = abil.begin(); i != abil.end();) {
+		const config &filter = (*i->first).child("filter_opponent");
+		const config &filter_attacker = (*i->first).child("filter_attacker");
+		const config &filter_defender = (*i->first).child("filter_defender");
+		bool abil_affect;//used for determine if abilities with special_affects_self or opponent must be erased of list
+		if(affect_self && !affect_other){
+			abil_affect = !special_affects_self((*i->first), attacker);
+		} else if(affect_other && !affect_self){
+			abil_affect = !special_affects_opponent((*i->first), attacker);
+		} else if(attacker && affect_other && affect_self){
+			abil_affect = (attacker  &&  apply_to == "attacker");
+		} else if(!attacker && affect_other && affect_self){
+			abil_affect = (!attacker  &&  apply_to == "defender");
+		}
+		bool fighter_filter = ability_apply_filter(un, up, ability, *i->first, loc, opp_loc, attacker, weapon, opp_weapon);
+		const std::string& active_on = (*i->first)["active_on"];
+		bool active_on_bool = !(active_on.empty() || (attacker && active_on == "offense") || (!attacker && active_on == "defense"));
+		if(!up &&(filter || (filter_attacker && !attacker) || (filter_defender && attacker))) {
+			return abiln;
```

Comment:
whats the point of this line? it seems like you abort the listing of all abilities if one of them has a [filter_opponent] and the there is no opponent which doesnt make sense, whether an ability has a [filter_opponent] tag must not have an influence on whether other abilities pass or not.

Reply:
@gfgtdf i have removed this

Reply:
i still see it

Reply:
oh i corrected a bug but i don't understand your comment, sorry.

Reply:
@gfgtdf i'm sorry, but this line is necessary for showing damage  and attack in unit bar when not 
fight selected, if abilite have [filter_opponent] or other then dmage showed must be without abilites or special calculation because the condition can't be matched.

Reply:
No, this cannot be correct fix you your mentioned problem, as i said before: With this **an [filter_defender] in one [damage] tag will prevent other [damage] tag from showing up even when they don't have a [filter_defender] subtag**

Reply:
@gfgtdf  this time , it is really fixed.